### DISCUSSION TITLE
Getting Malformed utf-8 character in laravel livewire in webview.

### DISCUSSION THREAD

**Discussion by sharpjsdev** [spatie/laravel-dashboard-twitter-tile#8] [Q&A]
i am getting error in application made with laravel livewire. getting issue of malformed utf-8 character when livewire event loadmore call. this issue only happened in webview. on in mobile view but api is same.

i search a lot but no solutions found.

**Comment by eldor** [comment]
Hi @sharpjsdev,
We found out that the package isn't multi-byte safe when using substr, which result in multi bytes getting cut off to early.

In [Tweet.php](src/Tweet.php) we changed line 94 from:

` $text = substr($text, $displayTextRange[0], $displayTextRange[1]);`

to:

` $text = mb_substr($text, $displayTextRange[0], $displayTextRange[1]);`

Which fixed our Malformed utf-8 character error.

Hope it helps you.

**Reply by freekmurze** [reply]
Feel free to PR that change 👍

**Reply by eldor** [reply]
#9 Here you go.
